compose/snippets/src/main/java/com/example/compose/snippets/animations/sharedelement/AnimatedVisibilitySharedElementSnippets.kt

```diff
@@ -0,0 +1,353 @@
+            columns = GridCells.Adaptive(150.dp),
+            verticalArrangement = Arrangement.spacedBy(8.dp),
+            horizontalArrangement = Arrangement.spacedBy(8.dp)
+        ) {
+            itemsIndexed(listSnacks) { index, snack ->
+                SnackItem(
+                    snack = snack,
+                    onClick = {
+                        selectedSnack = snack
+                    },
+                    visible = selectedSnack != snack
+                )
+            }
+        }
+
+        SnackEditDetails(
+            snack = selectedSnack,
+            onConfirmClick = {
+                selectedSnack = null
+            }
+        )
+    }
+}
+
+@Composable
+fun SharedTransitionScope.SnackItem(
+    snack: Snack,
+    visible: Boolean,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    AnimatedVisibility(
+        modifier = modifier,
+        visible = visible,
+        enter = fadeIn(animationSpec = animationSpec()) + scaleIn(
+            animationSpec()
+        ),
+        exit = fadeOut(animationSpec = animationSpec()) + scaleOut(
+            animationSpec()
+        )
+    ) {
+        Box(
+            modifier = Modifier
+                .sharedBounds(
+                    sharedContentState = rememberSharedContentState(key = "${snack.name}-bounds"),
+                    animatedVisibilityScope = this,
+                    boundsTransform = boundsTransition,
+                    clipInOverlayDuringTransition = OverlayClip(shapeForSharedElement)
+                )
+                .background(Color.White, shapeForSharedElement)
+                .clip(shapeForSharedElement)
+        ) {
+            SnackContents(
+                snack = snack,
+                modifier = Modifier.sharedElement(
+                    state = rememberSharedContentState(key = snack.name),
+                    animatedVisibilityScope = this@AnimatedVisibility,
+                    boundsTransform = boundsTransition,
+                ),
+                onClick = onClick
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalSharedTransitionApi::class, ExperimentalLayoutApi::class)
+@Preview
+@Composable
+private fun AnimatedVisibilitySharedElementShortenedExample() {
+    // [START android_compose_shared_elements_animated_visibility]
+    var selectedSnack by remember { mutableStateOf<Snack?>(null) }
+
+    SharedTransitionLayout(modifier = Modifier.fillMaxSize()) {
+        LazyVerticalGrid(
+            // [START_EXCLUDE]
+            modifier = Modifier
+                .fillMaxSize()
+                .background(Color.LightGray.copy(alpha = 0.5f))
+                .padding(16.dp),
+            columns = GridCells.Adaptive(150.dp),
+            horizontalArrangement = Arrangement.spacedBy(8.dp),
+            verticalArrangement = Arrangement.spacedBy(8.dp)
+            // [END_EXCLUDE]
+        ) {
+            itemsIndexed(listSnacks) { index, snack ->
+                AnimatedVisibility(
```

Comment:
In a real world use case, I imagine two common use cases for the item after it transitioned to the Details via shared element:

1. Devs may want the item to no longer take space in the list, which means other items would move in. I would expect animateItem to do that once lookahead is properly supported in LazyGrid.

or 2. The item that disappeared in the original list may have a place holder holding its place, so that no movements from other items are needed. This could be achieved by putting AV in a box, with AV's sibling observe the size of _AV's content_ via onSizeChanged, and setting itself to that size. 

For now, since 1 isn't feasible yet, could we try 2 to ensure continuity?

Reply:
BTW, LazyColumn has better support of lookahead.

We could consider showing an example of AVs in LazyColumn. That way we could set the exit on the AV to shrinkVertically to achieve the use case #1 mentioned above.

Reply:
done - switched to lazy column